### DISCUSSION TITLE
Using a three byte Emoji unicode character as a PasswordChar in a text box crashes the app.

### DISCUSSION THREAD

**Discussion by LaurentInSeattle** [AvaloniaUI/Avalonia#17554] [Q&A] [answered]
On Windows 10 Avalonia 11.1.4

This crashes at runtime: 
PasswordChar="&#x1f3cd;"	

Exception thrown: 'System.FormatException' in System.Private.CoreLib.dll
An exception of type 'System.FormatException' occurred in System.Private.CoreLib.dll but was not handled in user code
String must be exactly one character long.

This works: 
PasswordChar="&#x29BF;"

Not really an issue for me... And looks like a .Net for Windows bug...

**Comment by maxkatz6** [comment, marked as answer]
'🏍' is not really a character, instead it's a pair of two characters, and could be more.

Since .NET doesn't allow setting such emoji as a `char`, we won't be able to support them. Unless we change API to use `string` instead. I.e. `string PasswordMask` or something.

```
> '🏍'
< (1,19): error CS1012: Too many characters in character literal

> "🏍".ToCharArray()
< char[2] { '\ud83c', '\udfcd' }
```

**Reply by LaurentInSeattle** [reply]
Yes, TIL, .Net does not fully support Unicode, is limited to UTF-16.
